lib/Pipes/PipelineManager.cpp

```diff
@@ -162,26 +172,161 @@ checkComponentsVersion(const revng::DirectoryPath &ExecutionDirectory,
   return MaybeWritableFile.get()->commit();
 }
 
-static llvm::Expected<Runner>
-setUpPipeline(pipeline::Context &PipelineContext,
-              Loader &Loader,
-              llvm::ArrayRef<std::string> TextPipelines,
-              const revng::DirectoryPath &ExecutionDirectory) {
-  auto MaybePipeline = Loader.load(TextPipelines);
+static llvm::Expected<revng::FilePath>
+getBackupFilePath(const revng::FilePath &Path) {
+  // Keep trying to get a non-occupied backup file path. The timestamp is
+  // precise down to milliseconds, so it is low probability this takes
+  // more than one iteration.
+  while (true) {
+    const uint64_t UnixTime = revng::getEpochInMilliseconds();
+    auto BackupFilePath = Path.addExtension(std::to_string(UnixTime));
+    auto MaybeExists = BackupFilePath.exists();
+
+    if (not MaybeExists)
+      return MaybeExists.takeError();
+
+    if (not MaybeExists.get())
+      return BackupFilePath;
+  }
+}
+
+static llvm::Error migrateModelFile(const revng::FilePath &ModelFile) {
+  TemporaryFile InputTemporaryFile("revng-model-migration-input", "yml");
+  TemporaryFile OutputTemporaryFile("revng-model-migration-output", "yml");
+
+  revng::FilePath
+    InputFile = revng::FilePath::fromLocalStorage(InputTemporaryFile.path());
+  if (auto Error = ModelFile.copyTo(InputFile); !!Error)
+    return Error;
+
+  {
+    int ReturnCode = ::Runner.run("revng",
+                                  { "model",
+                                    "migrate",
+                                    InputTemporaryFile.path().str(),
+                                    "--output",
+                                    OutputTemporaryFile.path().str() });
+
+    if (ReturnCode != 0)
+      return revng::createError("Error encountered while running migrations, "
+                                "process returned: %d",
+                                ReturnCode);
+  }
+
+  revng::FilePath
+    OutputFile = revng::FilePath::fromLocalStorage(OutputTemporaryFile.path());
```

Comment:
Use `auto`. Also consider adding `using revng::FilePath` or even `using namespace revng` in the `.cpp` file.

Review all usages.

Reply:
I think this and previous occurrences can be made `auto` since the type is kinda obvious from the right hand side of the assignment. I don't think `auto` is warranted in other cases though as types are not exactly clear and `check-conventions` is happy. I would even consider removing some `auto`s in this commit where RHS is `File.exists` or `Directory.getFile()`.

I would add `using revng::FilePath`. I'm not sure about `using namespace revng` though, as I remember being confused by the similar-sounding error utilities in `revng` and `llvm`.

What do you think?

Reply:
You can use `auto` if the resulting type is obvious and on the same line. Basically with `dyn_cast` and factory methods (such as `FilePath::fromLocalStorage`).

For the others, add `using revng::FilePath`.

Reply:
(Sorry for disappearing too long).

Now we're `using revng::FilePath`. I looked into other occurences and put `auto` where appropriate as discussed.